### PR TITLE
Fixing error thrown when field is of type "Function"

### PR DESCRIPTION
**Pull request by jakefurler**
Please:
- [x] Do not include the compiled `.js`, `.js.map`, or `.d.ts` in your pull request as it makes it harder to merge your changes. 
- [x] Make your pull request atomic, fixing one issue at a time unless there are many relevant issues that cannot be decoupled.
- [x] Provide a test case & update the documentation in the `Readme.md`

### CODE REVIEW

File: typescript-json-schema.ts [hunk 1]
```diff
@@ -543,29 +543,31 @@ export class JsonSchemaGenerator {
...
-                } else {
-                    definition.type = "array";
-                    definition.items = def;
+                    if (isStringIndexed) {
+                        definition.type = "object";
+                        definition.additionalProperties = def;
+                    } else {
+                        definition.type = "array";
+                        definition.items = def;
+                    }
```
Comment (jakefurler) [inline typescript-json-schema.ts]: the diff isn't very helpful, I just wrapped it all in `if (clazz.members)`. This was present before, but removed in https://github.com/YousefED/typescript-json-schema/pull/117/files

**Comment by jakefurler** [comment]
The failing test is a timeout issue.

If you're happy with this change, I'd very much appreciate a new release once it's merged if possible :)

**Comment by domoritz** [comment]
Done as 0.16

**Comment by jakefurler** [comment]
Thanks @domoritz !

**Pull request merged**